pkg/operator/controller/ingress/status.go

```diff
@@ -102,6 +105,73 @@ func computeIngressTLSProfile(oldProfile *configv1.TLSProfileSpec, deployment *a
 	return newProfile
 }
 
+// computeDeploymentPodsScheduledCondition computes the ingress controller's
+// current PodsScheduled status condition state by inspecting the PodScheduled
+// conditions of the pods associated with the deployment.
+func computeDeploymentPodsScheduledCondition(deployment *appsv1.Deployment, pods []corev1.Pod) operatorv1.OperatorCondition {
+	unscheduled := make(map[*corev1.Pod]corev1.PodCondition)
+	selector, err := metav1.LabelSelectorAsSelector(deployment.Spec.Selector)
+	if err != nil || selector.Empty() {
+		return operatorv1.OperatorCondition{
```

Comment:
Could you add a unit test that reaches this condition?

Reply:
Done.